### ISSUE TITLE
Semcheck regression when accessing a static parameter in proc

### ISSUE THREAD

**Issue by mratsim** [nim-lang/Nim#14136]
The following used to compile in Nim v1.2.0

```Nim
type
  MDigest*[bits: static[int]] = object
    ## Message digest type
    data*: array[bits div 8, byte]

  Sha2Context*[bits: static[int],
               bsize: static[int],
               T: uint32|uint64] = object
    count: array[2, T]
    state: array[8, T]
    buffer: array[bsize, byte]

  sha256* = Sha2Context[256, 64, uint32]

template hmacSizeBlock*(h: typedesc): int =
  when (h is Sha2Context):
    int(h.bsize)
  else:
    {.fatal: "Choosen hash primitive is not yet supported!".}

type
  HMAC*[HashType] = object
    ## HMAC context object.
    mdctx: HashType
    opadctx: HashType
    ipad: array[HashType.hmacSizeBlock, byte]
    opad: array[HashType.hmacSizeBlock, byte]

func hkdfExtract*[T;S,I: char|byte](ctx: var HMAC[T],
                     prk: var MDigest[T.bits], # <------- error here "Error: type expected"
                     salt: openArray[S],
                     ikm: openArray[I]
                    ) =
  discard

var ctx: HMAC[sha256]
var prk: MDigest[sha256.bits]
let salt = [byte 0x00, 0x01, 0x02]
let ikm = "CompletelyRandomInput"

ctx.hkdfExtract(prk, salt, ikm)
```

This is based on nimcrypto and nim-blscurve code. Unfortunately nim-blscurve was deactivated from CI due to a dependency for testing on NimYAML which was broken due to YAML upstream deleting commits from their repo ... (https://github.com/status-im/nim-blscurve/issues/39, https://github.com/flyx/NimYAML/issues/77)

**Comment by ghost** [comment]
Git bisect says:
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[10eabec6d442cb32c2c74f7c56cc9fff3be40104] fix #12864 static params were mutating arg types during sigmatch; fix #12713 ; refs #13529 (#13976)

**Comment by Araq** [comment]
Confirmed.
